Sources/BrowserServicesKit/Autofill/AutofillUserScript.swift

```diff
@@ -19,16 +19,70 @@
 
 import WebKit
 
+#if !os(iOS)
+/// Handles calls from the top Autofill context to the overlay
+public protocol TopOverlayAutofillUserScriptDelegate: AnyObject {
+    /// Provides a size that the overlay should be resized to
+    func requestResizeToSize(width: CGFloat, height: CGFloat)
+}
+
+/// Is used by the top Autofill to reference into the child autofill
+public protocol AutofillMessagingToChildDelegate {
+    /// Represents the last tab host, is used to verify messages origin from and selecting of credentials
+    var lastOpenHost: String? { get }
+    /// Handles filling the credentials back from the top autofill into the child
+    func messageSelectedCredential(_ data: [String: String], _ configType: String)
+    /// Closes the overlay
+    func close()
+}
+
+/// Handles calls from the child Autofill context to the overlay.
+public protocol ChildOverlayAutofillUserScriptDelegate: AnyObject {
+    var view: NSView { get }
+    /// Closes the overlay
+    func autofillCloseOverlay(_ autofillUserScript: AutofillMessagingToChildDelegate?)
+    /// Opens the overlay
+    func autofillDisplayOverlay(_ messageInterface: AutofillMessagingToChildDelegate, of view: NSView, serializedInputContext: String, click: CGPoint, inputPosition: CGRect)
+}
+#endif
+
 public class AutofillUserScript: NSObject, UserScript {
```

Comment:
Changes to this script seem to lack separation of concerns, I think we should have two separate classes for this:
- WebsiteAutofillUserScript
- OverlayAutofillUserScript

Each having a separate delegates, responsibilities and set of supported messages. With that, we probably could get rid of iOS-related conditions as iOS would just not use the Overlay one. 

In the current shape, it is not clear for me if Im supposed to call `autofillInterfaceToChild.close()` or `currentOverlayTab.autofillCloseOverlay(self)` from the Autofill Script and have to carefully think about message flow to figure that out.

Above will also improve delegate method naming to `WebsiteAutofillUserScriptDelegate` and `OverlayAutofillUserScriptDelegate`

Reply:
Done :)